src/components/ErrorBoundary.test.tsx

```
@@ -54,4 +66,157 @@ describe('ErrorBoundary', () => {
+    expect(consoleErrorSpy).toHaveBeenCalledWith('Component stack:', expect.any(String));
+  });
+
+  it('catches errors in event handlers within children', () => {
+    const ButtonWithError = () => {
+      const [hasError, setHasError] = useState(false);
+
+      if (hasError) {
+        throw new Error('Event handler error');
+      }
+
+      return <button onClick={() => setHasError(true)}>Click to error</button>;
```

Biome (2.3.14)

[error] Provide an explicit type prop for the button element.

The default type of a button is submit, which causes the submission of a form when placed inside a `form` element. This is likely not the behaviour that you want inside a React application.
Allowed button types are: submit, button or reset

(lint/a11y/useButtonType)